src/routes/snippets.rs

```diff
@@ -282,9 +282,10 @@ pub fn get_raw_snippet(
 }
 
 #[get("/snippets/<id>", rank = 2)]
-pub fn get_snippet(
-    storage: State<Box<dyn Storage>>,
+pub async fn get_snippet(
+    storage: &State<Box<dyn Storage>>,
     id: String,
+    _content_type: &NegotiatedContentType,
```

Comment:
Just curious, why it wasn't necessary before but became necessary now?

Reply:
Great question!

IIRC, this is because `Responder` is not an `async` trait, and so we can no longer run request guard logic in the `Responder` implementation for `Output` (where we figure out what the format of the response should be). So we need to somehow trigger this request guard in every route. I agree that it looks weird, because the whole point of having `Output` was so that we can do it transparently, but alas :( 

Alternatively, we can make it even more explicit and pass the content type value when constructing an `Output` instance.